Zend/zend_builtin_functions.c

```diff
@@ -1505,6 +1505,20 @@ ZEND_FUNCTION(get_resource_type)
 }
 /* }}} */
 
+/* {{{ proto int get_resource_id(resource res)
+   Get the resource ID for a given resource */
+ZEND_FUNCTION(get_resource_id)
```

Comment:
I guess resource-ids will be re-used across request boundaries? if so, should be mentioned somewhere

Reply:
I'd agree it should be documented. Obviously, this RFC doesn't change the existing behavior, and the same note also applies to (int)$resource.
(e.g. "currently, resource ids are unique for the lifetime of execution (e.g. an http request in a web server), but this may change, because it's limited to `2**32` on 32-bit platforms)

I guess part of the issue is that it's using an associative array from integers to strings, and that zend_hash_index_del doesn't reset the last array pointer. I guess the only place where this could matter is that long-running 32-bit applications that create a lot of resources would possibly be affected by this (only able to create 4 billion ids in a lifetime for sockets/files, unless I misunderstood how this works), but for 64-bit applications, it doesn't matter.

Alternately, I guess it'd be possible to maintain a "free list" like what objects do so that resource ids can be reused.

```
php > for ($i = 0; $i < 3; $i++) { $fp = fopen("test$i.out", "w"); echo (int)$fp, "\n"; fclose($fp); $fp = null; }
2
3
4
php > $x = ['a', 'b'];
php > unset($x[1]);
php > $x[] = 'c';
php > var_export($x);
array (
  0 => 'a',
  2 => 'c',
)
```

```
ZEND_API void ZEND_FASTCALL zend_list_free(zend_resource *res)
{
	ZEND_ASSERT(GC_REFCOUNT(res) == 0);
	zend_hash_index_del(&EG(regular_list), res->handle);
}
````

Reply:
@TysonAndre It's a known issue, see https://bugs.php.net/bug.php?id=47396.